File: src/leetcode/count-of-matches-in-tournament/count-of-matches-in-tournament.js
/**
 * Leetcode #2006
 * Difficulty: Easy
 * https://leetcode.com/problems/count-number-of-pairs-with-absolute-difference-k/
 */
/**
 * @param {number} n
 * @return {number}
 */
export const numberOfMatches = n => {
  let matches = 0;

  while (n > 1) {
    const value = n / 2;
    matches += Math.floor(value);
    n = Math.ceil(value);
  }

  return matches;
};

// const numberOfMatches = n => n - 1;

// const numberOfMatches = n => {
//   let matches = 0;
//
//   while (true) {
//     const value = n / 2;
//
//     if (value < 1) {
//       return matches;
//     }
//
//     matches += Math.floor(value);
//     n = Math.ceil(value);
//   }
// };

File: src/leetcode/count-of-matches-in-tournament/count-of-matches-in-tournament.rs
impl Solution {
    pub fn number_of_matches(mut n: i32) -> i32 {
        let mut matches = 0;
        let mut temp: f64 = n as f64;

        loop {
            let value = temp / 2.0;

            if (value < 1.0) {
                break;
            }


            matches += value.floor() as i32;
            temp = value.ceil();
        }

        matches
    }
}